File: backend/api/schemas/document.py
"""Pydantic schemas for documents."""

from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field


class DocumentResponse(BaseModel):
    """Schema for document response."""
    id: int
    filename: str
    original_filename: str
    file_type: str
    file_size: int
    chunks_count: Optional[int] = None
    uploaded_at: datetime
    is_ingested: bool
    ingestion_status: str
    error_message: Optional[str] = None

    class Config:
        from_attributes = True


class DocumentListResponse(BaseModel):
    """Schema for listing documents."""
    documents: list[DocumentResponse]
    total: int


class DocumentStatsResponse(BaseModel):
    """Schema for document statistics."""
    total_documents: int
    total_size_mb: float
    total_chunks: int
    by_type: dict[str, int]

File: frontend/src/types/document.ts
export interface DocumentType {
  id: number
  filename: string
  original_filename: string
  file_type: string
  file_size: number
  chunks_count: number | null
  uploaded_at: string
  is_ingested: boolean
  ingestion_status: 'pending' | 'processing' | 'completed' | 'failed'
  error_message: string | null
}

export interface DocumentStats {
  total_documents: number
  total_size_mb: number
  total_chunks: number
  by_type: Record<string, number>
}
